addons/mail/static/src/models/messaging/messaging.js

```diff
@@ -16,18 +16,69 @@ function factory(dependencies) {
             return super._willDelete(...arguments);
         }
 
+        /**
+         * Starts messaging and related records.
+         */
+        async start() {
+            this._handleGlobalWindowFocus = this._handleGlobalWindowFocus.bind(this);
+            this.env.services['bus_service'].on('window_focus', null, this._handleGlobalWindowFocus);
+            await this.async(() => this.initializer.start());
+            this.notificationHandler.start();
+            this.update({ isInitialized: true });
+        }
+
         //----------------------------------------------------------------------
         // Public
         //----------------------------------------------------------------------
 
         /**
-         * Open the form view of the record with provided id and model.
+         * Gets the canonical chat with the provided person and returns it.
+         *
+         * If a chat is not appropriate a notification is displayed instead.
+         *
+         * @param {Object} param0
+         * @param {integer} [param0.partnerId]
+         * @param {integer} [param0.userId]
+         * @param {Object} [options]
+         * @returns {mail.thread|undefined}
+         */
+        async getCanonicalChat({ partnerId, userId }) {
```

Comment:
would simply use `getChat`, "canonical" is scary for no apparent reason, so would drop it

Reply:
ok